### PR TITLE
Plugins

### PR DESCRIPTION
**Pull request by 0xAF**

### CODE REVIEW

File: htdocs/plugins.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ * You should load your plugins in "plugins/{type}/init.js",
+ * where {type} is 'receiver' or 'map'.
+ * See the init.js.sample.
+ * And check the "plugins/{type}/example" folder for example plugin.
+ */
+
+// Wait for the page to load, then load the plugins.
+$(document).ready(function () {
+  // detect which plugins to load
```
Comment (luarvique) [inline htdocs/plugins.js:12]: This is a really hacky way to determine if we are in receiver or the map. Please, consider a simpler, more straightforward way.

File: htdocs/plugins.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  // check if we are loading an url
+  if (name.toLowerCase().startsWith('https://') || name.toLowerCase().startsWith('http://')) {
+    path = name;
+    name = path.split('/').pop().split('.').slice(0, -1);
+    path = path.split('/').slice(0, -1).join('/') + '/';
+    remote = true;
+  }
+
+  // ignore plugins with same name as our methods
+  if (name === "load" || name === "init" || name === "isLoaded") {
```
Comment (luarvique) [inline htdocs/plugins.js]: This looks like another hack. What happens if you add another function to Plugins? Is there a way to do this simply and logically?

File: htdocs/plugins.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+          return;
+        }
+      }
+
+      // check if plugin has set the 'no_css', otherwise, load the plugin css style
+      if (!('no_css' in Plugins[name])) {
+        Plugins._load_style(style_src)
+          .then(function () {
+            Plugins[name]._style_loaded = style_src;
+            Plugins._debug(`PLUGINS: ${remote ? '[remote]' : '[local]'} '${name}' loaded.`);
```
Comment (luarvique) [inline htdocs/plugins.js]: You can generate the "PLUGINS: ..." banner automatically, by a separate function given just plugin name.

Comment (0xAF) [reply htdocs/plugins.js]: yes... I just added the _debug function right before the commit and I did not reworked that... will do.

File: htdocs/plugins.js (hunk 1 above)
Comment (0xAF) [reply htdocs/plugins.js:12]: I did not wanted to call the init() with an argument (receiver/map) from the index.html directly. I tried to keep the main code clean as much as I could. I will think about another solution.

File: htdocs/plugins.js (hunk 2 above)
Comment (0xAF) [reply htdocs/plugins.js]: This check could be dropped actually. I've implemented a check if the plugin is already loaded, so we will not load it again. That check can handle this case too. Will rework this.

**Pull request merged**